client/js/chart/index.js

```diff
@@ -30,63 +33,46 @@ export function createChart(el, rankingHistory, rankingDates) {
   const seriesElements = [];
   const rankingHistoryEntries = [...rankingHistory];
 
+  // to organize data into rows to speed up hit-testing
+  const rows = [];
+
+  for (let row = 0; row < rankingDates.length; row++) {
+    rows.push([]);
+  }
+
   rankingHistoryEntries.forEach(([, history], index) => {
     const hue = index * 15 % 360;
```

Comment:
add some constants so `15` and `360` don't seem like magic nums

Reply:
no lol you don't need to abstract every number into a constant
`360` is the number of degrees in a circle i swear if nick tells me to make that a constant

Reply:
:)

Reply:
You don't necessarily need a const here. In fact, a comment would be more readable. For something like:
```
const MS_IN_DAY = 24 * 60 * 60 * 1000;
```
you use the const name to indicate that the magic numbers that are about to show up have the context of time and ultimately are measured in milliseconds. Since your order of operations don't let you store this equation in a const easily, a comment like `Histories should be assigned a rotating set of colors that are 15 degrees apart on the color wheel. This will give us 24 unique colors.` is preferred. This tells us the units and intention of our magic numbers by explaining the outcome we want.